sites/docs/src/content/docs/tutorials/tests_and_test_data/components/04_testing_modules.md

```diff
@@ -0,0 +1,550 @@
+---
+title: "4. Testing Modules"
+subtitle: Testing individual nf-core modules with real examples from nf-core/methylseq
+weight: 40
+---
+
+## Module Test Structure
+
+A typical nf-core module test follows established patterns. Here's the structure used in nf-core/methylseq for FastQC:
+
+```groovy
+nextflow_process {
+    name "Test Process FASTQC"
+    script "../main.nf"
+    process "FASTQC"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "fastqc"
+
+    test("sarscov2 single-end [fastq]") {
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id: 'test', single_end:true ],
+                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true) ]
+                ])
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                // NOTE The report contains the date inside it, which means that the md5sum is stable per day, but not longer than that. So you can't md5sum it.
+                // looks like this: <div id="header_filename">Mon 2 Oct 2023<br/>test.gz</div>
+                // https://github.com/nf-core/modules/pull/3903#issuecomment-1743620039
+                { assert process.out.html[0][1] ==~ ".*/test_fastqc.html" },
+                { assert process.out.zip[0][1] ==~ ".*/test_fastqc.zip" },
+                { assert path(process.out.html[0][1]).text.contains("<tr><td>File type</td><td>Conventional base calls</td></tr>") },
+                { assert snapshot(process.out.versions).match() }
+            )
+        }
+    }
+}
+```
+
+## Essential Module Testing Components
+
+### 1. Standard Test Tags
+
+All nf-core modules use consistent tagging:
+
+```groovy
+tag "modules"
+tag "modules_nfcore"
+tag "fastqc"  // Tool-specific tag
+```
+
+### 2. Meta Map Structure
+
+Meta maps follow nf-core conventions:
+
+```groovy
+// Single-end reads
+[ id: 'test', single_end: true ]
+
+// Paired-end reads  
+[ id: 'test', single_end: false ]
+```
+
+### 3. Test Data Usage
+
+Use nf-core test datasets with `checkIfExists: true`:
+
+```groovy
+file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
+```
+
+## Real Module Testing Examples
+
+### FastQC Module - Multiple Input Types
```

Comment:
You've jumped ahead here, you explain tags but what about the other components of the test file?